<?xml version="1.0"?>
<!--
Copyright (c) 2020, STEREOLABS.

All rights reserved.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->
<launch>
    <arg name="svo_file"             default="" /> <!-- <arg name="svo_file" default="path/to/svo/file.svo"> -->
    <arg name="stream"               default="" /> <!-- <arg name="stream" default="<ip_address>:<port>"> -->

    <arg name="node_name_1"            default="zed_node_A" />
    <arg name="camera_model"         default="zedm" /> <!-- 'zed' or 'zedm' -->
    <arg name="publish_urdf"         default="true" />

    <arg name="camera_name_1"          default="zedm_A" />

    <arg name="base_frame"           default="base_link" />

    <arg name="cam_pos_x"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_y"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_z"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_roll"              default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_pitch"             default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_yaw"               default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->

    <arg name="node_name_2"            default="zed_node_B" />
    <arg name="camera_name_2"          default="zedm_B" />

    <arg name="camera_sn_1"             default="10029656" />
    <arg name="camera_sn_2"             default="17233420" />


    <group ns="$(arg camera_name_1)">
        <include file="$(find tauv_vehicle)/launch/zed/include/zed_camera.launch.xml">
            <arg name="camera_name"         value="$(arg camera_name_1)" />
            <arg name="svo_file"            value="$(arg svo_file)" />
            <arg name="stream"              value="$(arg stream)" />
            <arg name="node_name"           value="$(arg node_name_1)" />
            <arg name="camera_model"        value="$(arg camera_model)" />
            <arg name="base_frame"          value="$(arg base_frame)" />
            <arg name="publish_urdf"        value="$(arg publish_urdf)" />
            <arg name="camera_sn"           value="$(arg camera_sn_1)" />
            <arg name="cam_pos_x"           value="$(arg cam_pos_x)" />
            <arg name="cam_pos_y"           value="$(arg cam_pos_y)" />
            <arg name="cam_pos_z"           value="$(arg cam_pos_z)" />
            <arg name="cam_roll"            value="$(arg cam_roll)" />
            <arg name="cam_pitch"           value="$(arg cam_pitch)" />
            <arg name="cam_yaw"             value="$(arg cam_yaw)" />
            <arg name="camera_id"           value="0" />
            <!-- <remap from="left/image_rect_color" to="/cameras/footage_merged"/> -->
        </include>
        <node name="cam_b_remapper" pkg="topic_tools" type="relay" args="zed_node_A/left/image_rect_color /cameras/merged"/>
        <!-- <remap from="zed_node_A/left/image_rect_color" to="/cameras/footage_merged"/> -->
    </group>
    <group ns="$(arg camera_name_2)">
        <include file="$(find tauv_vehicle)/launch/zed/include/zed_camera.launch.xml">
            <arg name="camera_name"         value="$(arg camera_name_2)" />
            <arg name="svo_file"            value="$(arg svo_file)" />
            <arg name="stream"              value="$(arg stream)" />
            <arg name="node_name"           value="$(arg node_name_2)" />
            <arg name="camera_model"        value="$(arg camera_model)" />
            <arg name="base_frame"          value="$(arg base_frame)" />
            <arg name="publish_urdf"        value="$(arg publish_urdf)" />
            <arg name="camera_sn"           value="$(arg camera_sn_2)" />
            <arg name="cam_pos_x"           value="$(arg cam_pos_x)" />
            <arg name="cam_pos_y"           value="$(arg cam_pos_y)" />
            <arg name="cam_pos_z"           value="$(arg cam_pos_z)" />
            <arg name="cam_roll"            value="$(arg cam_roll)" />
            <arg name="cam_pitch"           value="$(arg cam_pitch)" />
            <arg name="cam_yaw"             value="$(arg cam_yaw)" />
            <arg name="camera_id"           value="1" />
        </include>
        <node name="cam_b_remapper" pkg="topic_tools" type="relay" args="zed_node_B/left/image_rect_color /cameras/merged"/>
        <!-- <remap from="zed_node_B/left/image_rect_color" to="/cameras/footage_merged"/> -->
        <!-- <remap from="left/image_rect_color" to="/cameras/footage_merged"/> -->
    </group>
</launch>
